### PR TITLE
kv/LevelDBStore, FileStore, MonDBStore: simpler code for single-key fetches

### PR DESCRIPTION
**Pull request by branch-predictor**
Often there's a need to fetch value for a specified, single key. Publish
a new method in LevelDB that takes key and prefix and puts just value at
specified address, without the need for interim keysets or resultsets.
Make use of that method in MonitorDBStore and LevelDB (not all cases, only
the most often called ones).

Signed-off-by: Piotr Dałek piotr.dalek@ts.fujitsu.com

### CODE REVIEW

File: src/kv/LevelDBStore.cc [hunk 1]
```diff
@@ -218,6 +218,22 @@ int LevelDBStore::get(
...
 
+int LevelDBStore::get(const string &prefix, 
+		  const string &key,
+		  bufferlist *value)
+{
+  utime_t start = ceph_clock_now(g_ceph_context);
+  KeyValueDB::Iterator it = get_iterator(prefix);
+  it->lower_bound(key);
+  if (it->valid() && it->key() == key) {
+      *value = it->value();
```
Comment (liewegas) [inline src/kv/LevelDBStore.cc]: i think we need to else return -ENOENT.  can probably mostly cut&paste from here:

https://github.com/liewegas/ceph/blob/wip-kv/src/kv/RocksDBStore.cc#L296

File: src/mon/MonitorDBStore.h [hunk 2]
```diff
@@ -507,14 +507,12 @@ class MonitorDBStore
...
-
-    db->get(prefix, k, &out);
-    if (out.empty())
+    
+    bufferlist outbl;
+    db->get(prefix, key, &outbl);
+    if (outbl.length() == 0) 
       return -ENOENT;
-    bl.append(out[key]);
+    bl.append(outbl);
```
Comment (liewegas) [inline src/mon/MonitorDBStore.h:515]: yes

File: src/os/DBObjectMap.cc [hunk 3]
```diff
@@ -1089,17 +1089,16 @@ DBObjectMap::Header DBObjectMap::_lookup_map_header(
...
+  int r = db->get(HOBJECT_TO_SEQ, map_header_key(oid), &out);
+  if (r < 0 || out.length()==0) {
     delete header;
     return Header();
   }
 
   Header ret(header, RemoveOnDelete(this));
-  bufferlist::iterator iter = out.begin()->second.begin();
+  bufferlist::iterator iter = out.begin();
+
```
Comment (liewegas) [inline src/os/DBObjectMap.cc:1101]: yes

let's break these 2 other hunks into separate patches.  they work already since there is the generic single-key get wrapper in KeyValueDB

**Comment by branch-predictor** [comment]
@liewegas done.
